python/lsst/pipe/tasks/mkTransientFakesCatalog.py

```diff
@@ -0,0 +1,345 @@
+        doc="WCS information for the input exposure.",
+        name="jointcal_wcs",
+        storageClass="Wcs",
+        dimensions=("Tract", "SkyMap", "Instrument", "Visit", "Detector")
+    )
+
+    photoCalib = cT.Input(
+        doc="Calib information for the input exposure.",
+        name="jointcal_photoCalib",
+        storageClass="PhotoCalib",
+        dimensions=("Tract", "SkyMap", "Instrument", "Visit", "Detector")
+    )
+
+    icSourceCat = cT.Input(
+        doc="Catalog of calibration sources",
+        name="icSrc",
+        storageClass="sourceCatalog",
+        dimensions=("tract", "skymap", "instrument", "visit", "detector")
+    )
+
+    sfdSourceCat = cT.Input(
+        doc="Catalog of calibration sources",
+        name="src",
+        storageClass="sourceCatalog",
+        dimensions=("tract", "skymap", "instrument", "visit", "detector")
+    )
+
+    outputExposure = cT.Output(
+        doc="Exposure with fake sources added.",
+        name="fakes_calexp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector")
+    )
+
+    outputCat = cT.Output(
+        doc="Source catalog produced in calibrate task with fakes also measured.",
+        name="fakes_src",
+        storageClass="SourceCatalog",
+        dimensions=("instrument", "visit", "detector"),
+    )
+
+
+class MkTransientFakesCatalogConfig(PipelineTaskConfig,
+                                pipelineConnections=MkTransientFakesCatalogConnections):
+    """Config for inserting fake sources
+
+    Notes
+    -----
+    The default column names are those from the UW sims database.
+    """
+
+    useUpdatedCalibs = pexConfig.Field(
+        doc="Use updated calibs and wcs from jointcal?",
+        dtype=bool,
+        default=False,
+    )
+
+    coaddName = pexConfig.Field(
+        doc="The name of the type of coadd used",
+        dtype=str,
+        default="deep",
+    )
+
+    srcFieldsToCopy = pexConfig.ListField(
+        dtype=str,
+        default=("calib_photometry_reserved", "calib_photometry_used", "calib_astrometry_used",
+                 "calib_psf_candidate", "calib_psf_used", "calib_psf_reserved"),
+        doc=("Fields to copy from the `src` catalog to the output catalog "
+             "for matching sources Any missing fields will trigger a "
+             "RuntimeError exception.")
+    )
+
+    matchRadiusPix = pexConfig.Field(
+        dtype=float,
+        default=3,
+        doc=("Match radius for matching icSourceCat objects to sourceCat objects (pixels)"),
+    )
+
+    calibrate = pexConfig.ConfigurableField(target=CalibrateTask,
+                                            doc="The calibration task to use.")
+
+    insertFakes = pexConfig.ConfigurableField(target=InsertFakesTask,
+                                              doc="Configuration for the fake sources")
+
+    def setDefaults(self):
+        super().setDefaults()
+
+
+class MkTransientFakesCatalogTask(PipelineTask, CmdLineTask):
```

Comment:
Is this task name communicating what the task is doing? Looks to insert into an image and run calibrate to detect sources. The name currently reads like it's creating a catalog of fakes.